# U.S. Medical Insurance Costs

Hello guys! My name is Madhav Menon and I am 15 years old. This is my attempt at my first Portfolio Project, "US Medical Insurance Costs" from Codecademy as part of the Data Science career path

<u> Research Question </u>

How many people in the given dataset are smokers and how does their insurance charge differ from non-smokers?


<u> Goals </u>

1. To figure out the number of smokers in our given dataset.
2. To calculate their insurance charges and compare this to non-smokers to see how smoking can affect insurance costs.

<u> Actions </u>

I predict that there will be a drastic increase in insurance costs for smokers as opposed to non smokers, this would enable us to create some sort of promotional content such as flyers, advertisements etc. which will allow us to highlight the negative impacts of smoking on people's wallets and health. Cigarettes tend to be a demerit good but for some also tend to be an inelastic good so the economic analysis might be covered here as well. In conclusion, I hope that this information will enable us to reduce the number of smokers.

<u> Data </u>

I will need a dataset that contains mostly two columns, whether they are a smoker and their insurance costs. Luckily Codecademy has given us a dataset from Kaggle which gives more information such as their age, bmi, and sex.

<u> Analysis </u>

We will have to analyse the insurance costs of smokers and compare this to the insurance ocsts of non smokers. This will give a clear idea as to how smoking can affect the insurance costs of individuals. This may allow us to create awareness on the negative impacts of smoking. We can also conduct smaller studies such as the average age of smokers within the dataset, this may give us a rough idea of when people start smoking so necessary government action can be taken. We could also find out how many smokers have kids, this could lead to a whole different study of how parents' smoking can affect kids.



## Prologue

I start by importing Pandas because Pandas are essential for data analysis. The first line of code below was taken from StackOverflow as I wanted a more efficient way of opening the .csv file as opposed to using "with open()..."



In [2]:
import pandas

I then convert all the columns into lists that we can work with. I created a list called person_list because I wanted one list which contains all the data of a person. I also used a dictionary for convenience as it's easier to read information like this using a dictionary. 

In [65]:
data = pandas.read_csv("insurance.csv", header=0)

ages=list(data.age)
sexes=list(data.sex)
bmis=list(data.bmi)
num_children=list(data.children)
is_smoker=list(data.smoker)
regions=list(data.region)
charges=list(data.charges)

person_list=list(zip(ages,sexes,bmis,num_children,is_smoker,regions,charges)) 

dict_of_data={}
for i in range(len(ages)):
    dict_of_data[f"Person {i}"]={
                  "Age":ages[i],
                  "Sex":sexes[i],
                  "BMI":bmis[i],
                  "Number of children":num_children[i],
                  "Smoker": is_smoker[i],
                  "Region":regions[i],
                  "Insurance Cost":charges[i]}
print(dict_of_data)

{'Person 0': {'Age': 19, 'Sex': 'female', 'BMI': 27.9, 'Number of children': 0, 'Smoker': 'yes', 'Region': 'southwest', 'Insurance Cost': 16884.924}, 'Person 1': {'Age': 18, 'Sex': 'male', 'BMI': 33.77, 'Number of children': 1, 'Smoker': 'no', 'Region': 'southeast', 'Insurance Cost': 1725.5523}, 'Person 2': {'Age': 28, 'Sex': 'male', 'BMI': 33.0, 'Number of children': 3, 'Smoker': 'no', 'Region': 'southeast', 'Insurance Cost': 4449.462}, 'Person 3': {'Age': 33, 'Sex': 'male', 'BMI': 22.705, 'Number of children': 0, 'Smoker': 'no', 'Region': 'northwest', 'Insurance Cost': 21984.47061}, 'Person 4': {'Age': 32, 'Sex': 'male', 'BMI': 28.88, 'Number of children': 0, 'Smoker': 'no', 'Region': 'northwest', 'Insurance Cost': 3866.8552}, 'Person 5': {'Age': 31, 'Sex': 'female', 'BMI': 25.74, 'Number of children': 0, 'Smoker': 'no', 'Region': 'southeast', 'Insurance Cost': 3756.6216}, 'Person 6': {'Age': 46, 'Sex': 'female', 'BMI': 33.44, 'Number of children': 1, 'Smoker': 'no', 'Region': 'south

## Figure out the number of smokers in our dataset

I have now decided to find out the number of smokers in our given dataset. I must first find out the length of our dataset

In [8]:
length_of_dataset=len(ages)
print(length_of_dataset)

1338


We have a dataset of 1338 entries. I used the ages list just because it was the first list we created. In fact, we could have used any of the above lists as they are all unique entries and thus they would all have the same length. I now decided to count the number of smokers in our dataset

In [72]:
def count_smokers(is_smoker):
    count=0
    for smoker in is_smoker:
        if smoker=="yes":
            count+=1
        else:
            continue
    return count

number_of_smokers = count_smokers(is_smoker)
number_of_non_smokers=length_of_dataset-number_of_smokers
print(number_of_smokers)
print(number_of_non_smokers)



274
1064


In the above function, I initalised a variable called "count". It would increment by one for every "yes" entry in the is_smoker list. I then called the function and assigned it to the variable "number_of_smokers". Thus if we printed the variable, it would call the function each time. As you can see, 274 people in the dataset are smokers. I now want to calculate what percentage this is of the dataset

In [16]:
percentage_of_smokers = number_of_smokers/len(is_smoker)*100
print(percentage_of_smokers)

20.47832585949178


So far the calculations show that about 20% of the dataset smokes. 

### Average age of smokers

I now want to calculate the average age of the smokers in the dataset, this is a bit hard and this is where the master dictionary and list we created come in handy:




In [54]:
total_age = 0
for i in range(len(is_smoker)):
        if dict_of_data[f"Person {i}"]["Smoker"]=="yes":
            total_age+=dict_of_data[f"Person {i}"]["Age"]

average_age = total_age/number_of_smokers
print(average_age)


38.51459854014598


As you can see, the average age of smokers is about 39 years old. This is actually good news because the legal smoking age in the US is 21 and so there might be very few people in this dataset that are underaged smokers as the mean is closer to 40 and  not a smaller number. 

### Number of underaged smokers

In [58]:
number_of_underaged_smokers = 0
for i in range(len(is_smoker)):
    if dict_of_data[f"Person {i}"]["Smoker"]=="yes":
        if dict_of_data[f"Person {i}"]["Age"]<21:
            number_of_underaged_smokers+=1
print(number_of_underaged_smokers)
    

39


The code above shows that 39 of the 274 smokers are underaged. This is not good at all and calls for some serious government intervention. This is actually some good data because now we have numbers of substance to work with

In [60]:
percentage_of_underaged_smokers=number_of_underaged_smokers/number_of_smokers * 100
print(percentage_of_underaged_smokers)

percentage_of_underaged_smokers_dataset=number_of_underaged_smokers/len(is_smoker) * 100
print(percentage_of_underaged_smokers_dataset)


14.233576642335766
2.914798206278027


We know that around 14% of smokers are underaged and 3% of the entire dataset are underaged smokers as  well. This is because this should not even be 0.1% but unfortunately this isn't an ideal world. There should be some sort of advertising campaign from the government to prevent underaged smoking. They could also implement a better ID-tracking system of some sort so as to prevent fake IDs working successfuly. 

## Comparing insurance costs of smokers to non smokers

Now that some preliminary results have been calculated, let's get into the actual question, "How does the insurance cost of a smoker compare to the insurance cost of a non smoker?"

There are two methods to do this with the Python I know as of now:

1. Add up the insurance cost for all the smokers and non smokers and find the difference
2. Find the average insurance cost for all the smokers and non smokers and find the difference

There are some clear flaws with both the methods I have mentioned. These values take into account other factors such as BMI, number of children and their age which can also impact their insurance costs and thus this isn't a reliable test. However let's see how it goes

### Method 1

In [71]:
def insurance_cost_of_non_s (dict_of_data):
    charge = 0
    for i in range(len(charges)):
        if dict_of_data[f"Person {i}"]["Smoker"]=="no":
            charge+=dict_of_data[f"Person {i}"]["Insurance Cost"]
    return charge

total_insurance_cost_of_non_smokers = insurance_cost_of_non_s(dict_of_data)
print(total_insurance_cost_of_non_smokers)


def insurance_cost_of_s(dict_of_data):
    charge = 0
    for i in range(len(charges)):
        if dict_of_data[f"Person {i}"]["Smoker"]=="yes":
            charge+=dict_of_data[f"Person {i}"]["Insurance Cost"]
    return charge

total_insurance_cost_of_smokers = insurance_cost_of_s(dict_of_data)
print(total_insurance_cost_of_smokers)
    
difference_in_insurance_cost=total_insurance_cost_of_smokers - total_insurance_cost_of_non_smokers
print(difference_in_insurance_cost)

8974061.468918996
8781763.52184
-192297.9470789954


### Method 2

In [75]:
def average_cost_of_non_s(dict_of_data):
    charge = 0
    for i in range(len(charges)):
        if dict_of_data[f"Person {i}"]["Smoker"]=="no":
            charge+=dict_of_data[f"Person {i}"]["Insurance Cost"]
    return charge/(number_of_non_smokers)
            
average_insurance_cost_of_non_smokers=average_cost_of_non_s(dict_of_data)

def average_cost_of_s(dict_of_data):
    charge = 0
    for i in range(len(charges)):
        if dict_of_data[f"Person {i}"]["Smoker"]=="yes":
            charge+=dict_of_data[f"Person {i}"]["Insurance Cost"]
    return charge/(number_of_smokers)

average_insurance_cost_of_smokers = average_cost_of_s(dict_of_data)

print(average_insurance_cost_of_non_smokers)
print(average_insurance_cost_of_smokers)

8434.268297856199
32050.23183153285


### Explanation:



Clearly these two methods don't work for a variety of reasons. The first method didn't work because there are a lesser number of smokers as opposed to non smokers. Therefore there would be a higher frequency to add for non smokers which resulted in a higher total insurance cost. My explanation for method 2 will be more coherent. As there are more non smokers, there is a higher number you have to divide your charges by. This results in a small number as opposed to smokers who only have to divide by 274. 

### Final try:

I then decided to do one final thing. I take the insurance cost of one person by adding some stipulations. Their age is 18, they have a BMI of less than 27 and have no children. 

In [79]:
def new_insurance_cost_non_s(dict_of_data):
    charge=0
    for i in range(len(ages)):
        if dict_of_data[f"Person {i}"]["Smoker"]=="no" and dict_of_data[f"Person {i}"]["Age"]==18 and dict_of_data[f"Person {i}"]["BMI"]<30 and dict_of_data[f"Person {i}"]["Number of children"]==0:
            charge+=dict_of_data[f"Person {i}"]["Insurance Cost"]
    return charge

insurance_cost_of_non_smoker_with_restrictions = new_insurance_cost_non_s(dict_of_data)


def new_insurance_cost_non_s(dict_of_data):
    charge=0
    for i in range(len(ages)):
        if dict_of_data[f"Person {i}"]["Smoker"]=="yes" and dict_of_data[f"Person {i}"]["Age"]==18 and dict_of_data[f"Person {i}"]["BMI"]<30 and dict_of_data[f"Person {i}"]["Number of children"]==0:
            charge+=dict_of_data[f"Person {i}"]["Insurance Cost"]
    return charge

insurance_cost_of_smoker_with_restrictions = new_insurance_cost_non_s(dict_of_data)

print(insurance_cost_of_non_smoker_with_restrictions)
print(insurance_cost_of_smoker_with_restrictions)
difference=insurance_cost_of_smoker_with_restrictions-insurance_cost_of_non_smoker_with_restrictions 
print(difference)


31905.678019
43549.512
11643.833981000003


### Explanation

This data proved to be more conclusive. With the given stipulations, we were able to compare the insurance cost of the two individuals and found about a 12000 dollar difference in insurance cost between smokers and non smokers. While this data proves to be inconclusive without further results, it's a baseline for now. 

# Final Analysis

The initial research question was to find out how the insurance costs of smokers compared to the insurance costs of non smokers. I quickly found out after writing a few lines of code that my results cannot be reliable as we cannot have all the other variables constant. We were trying to measure the difference in insurance cost and we got the most reliable answer by adding many restrictions to our dataset. There are some clear flaws with this as we restricted the age to 18. How would this change for larger ages? The data showed that there was a 12000 dollar difference in insurance costs between smokers and non smokers following the givenr estrictions. This somewhat supported my hypothesis that smokers would have a larger insurance cost as opposed to non smokers. There are still many probing questions to conduct but instead I stumbled upon some pretty alarming information. I found out that 39 people in the dataset are underaged smokers. This calls for action from the relevant authorities as even having 1 underaged smoker is bad. However this isn't an ideal world but we have to try our best. 

# Thoughts

Overall I had lots of fun analysing this dataset. There was a lot of information and I usd a bit of help online. I was able to somewhat answer my questions but I know that it isn't very reliable. I would definitely analyse this again and look at a different aspect such as how the insurance cost for people with high BMIs differ to those with lower BMIs. I also had a lot of fun exploring Jupyter Notebook and I will definitely used this in further portfolio projects. 

### Appendix (Data collected):


Sample Size : 1338

Number of Smokers : 274

Percentage of smokers (Based on the entire dataset) : 20.47832585949178

Average age of a smoker : 38.51459854014598

Number of underaged smokers (Age<21) : 39

Percentage of underaged smokers (Based on number of smokers) : 14.233576642335766

Percentage of underaged smokers (Based on the entire dataset) : 2.914798206278027

Average insurance cost of non smokers : 8434.268297856199

Average insurance cost of smokers : 32050.23183153285

Insurance cost of non smoker with given restrictions (see above) : 31905.678019

Insurance cost of smoker with given restrictions (see above) : 43549.512

#### Resources:

1. https://www.kaggle.com/mirichoi0218/insurance
2. https://stackoverflow.com/questions/28283647/convert-csv-column-to-list
3. https://www.codecademy.com/paths/data-science/tracks/dscp-python-portfolio-project/modules/dscp-us-medical-insurance-costs/kanban_projects/us-medical-insurance-costs-portfolio-project





